app/controllers/heritages_controller.rb

```diff
@@ -151,6 +152,15 @@ def load_district
     @district = District.find_by!(name: params[:district_id])
   end
 
+  def load_district_by_param
+    if params[:district].present?
+      @district = District.find_by!(name: params[:district])
+      if @heritage.district.name != @district.name
+        raise ExceptionHandler::InternalServerError.new("Wrong district was passed")
+      end
+    end
```

Comment:
Nested if and reword the exception

```suggestion
    return if params[:district].empty?
    
    @district = District.find_by!(name: params[:district])
    if @heritage.district.name != @district.name
      raise ExceptionHandler::InternalServerError.new("The heritage #{@heritage.name} does not belong to district #{district.name}")
    end
```